jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -1603,6 +1604,71 @@ ecma_builtin_typedarray_prototype_find_index (ecma_value_t this_arg, /**< this a
   return ecma_builtin_typedarray_prototype_find_helper (this_arg, predicate, predicate_this_arg, false);
 } /* ecma_builtin_typedarray_prototype_find_index */
 
+/**
+ * The %TypedArray%.prototype object's 'indexOf' routine
+ *
+ * See also:
+ *          ECMA-262 v6, 22.2.3.13
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_bultin_typedarray_prototype_index_of (ecma_value_t this_arg, /**< this argument */
+                                           ecma_value_t arg1, /**< searchElement */
+                                           ecma_value_t arg2) /**< fromIndex */
+{
+  if (!ecma_is_typedarray (this_arg))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Argument 'this' is not a TypedArray."));
+  }
+
+  ecma_object_t *typedarray_p = ecma_get_object_from_value (this_arg);
+  uint32_t length = ecma_typedarray_get_length (typedarray_p);
+  uint8_t shift = ecma_typedarray_get_element_size_shift (typedarray_p);
+  uint8_t element_size = (uint8_t) (1 << shift);
+  uint32_t limit = length * element_size;
+
+  if (length == 0)
+  {
+    return ecma_make_integer_value (-1);
+  }
+
+  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+
+  ECMA_OP_TO_NUMBER_TRY_CATCH (arg_from_idx, arg2, ret_value);
+
+  uint32_t from_index = ecma_builtin_helper_array_index_normalize (arg_from_idx, length);
+
+  ecma_number_t found_index = ECMA_NUMBER_MINUS_ONE;
+
+  for (; from_index < limit && found_index < 0 && ecma_is_value_empty (ret_value); from_index += element_size)
+  {
+    ecma_string_t *from_index_str = ecma_new_ecma_string_from_uint32 (from_index);
+    ECMA_TRY_CATCH (element, ecma_op_object_find (typedarray_p, from_index_str), ret_value);
```

Comment:
AFAK incrementing the index and accessing the given element in this form only works with Uint8Array. Any "bigger" typedarray will have incorrect result.

example:
If the TypedArray used is Uint32Array the element_size will be 4. Thus when the operation is executed the `from_index` will be incremented by 4 in every case. This means that only every 4th element will be checked.

In addition using the `eca_op_objecto_find` is slow for TypedArrays. Check the function above ( https://github.com/jerryscript-project/jerryscript/pull/2938/files#diff-f81c66825ff663e22d61a8e1fb0fa8a9R1544 ) where the `ecma_get_typedarray_element` method is used to access the elements.